crates/bevy_reflect/src/impls/smol_str.rs

```diff
@@ -0,0 +1,167 @@
+use std::{
+    any::Any,
+    hash::{Hash, Hasher},
+};
+
+use smol_str::SmolStr;
+
+use crate::{
+    utility::{reflect_hasher, GenericTypePathCell, NonGenericTypeInfoCell},
+    FromReflect, FromType, GetTypeRegistration, Reflect, ReflectFromPtr, ReflectFromReflect,
+    ReflectMut, ReflectOwned, ReflectRef, TypeInfo, TypePath, TypeRegistration, Typed, ValueInfo,
+};
+
+impl Reflect for SmolStr {
+    fn type_name(&self) -> &str {
+        std::any::type_name::<Self>()
+    }
+
+    fn get_represented_type_info(&self) -> Option<&'static TypeInfo> {
+        Some(<Self as Typed>::type_info())
+    }
+
+    fn into_any(self: Box<Self>) -> Box<dyn Any> {
+        self
+    }
+
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn as_any_mut(&mut self) -> &mut dyn Any {
+        self
+    }
+
+    fn into_reflect(self: Box<Self>) -> Box<dyn Reflect> {
+        self
+    }
+
+    fn as_reflect(&self) -> &dyn Reflect {
+        self
+    }
+
+    fn as_reflect_mut(&mut self) -> &mut dyn Reflect {
+        self
+    }
+
+    fn apply(&mut self, value: &dyn Reflect) {
+        let value = value.as_any();
+
+        if let Some(value) = value.downcast_ref::<SmolStr>() {
```

Comment:
Nit: You shouldn't need to cast to `dyn Any`
```suggestion
        if let Some(value) = value.downcast_ref::<SmolStr>() {
```